rosbag_snapshot/src/snapshotter.cpp

```diff
@@ -606,10 +686,9 @@ int Snapshotter::run()
     status_timer_ = nh_.createTimer(options_.status_period_, &Snapshotter::publishStatus, this);
 
   // Start timer to poll ROS master for topics
-  if (options_.all_topics_)
```

Comment:
How would you feel about changing this to `if (options_.all_topics_ || !patterns_.empty())`